src/controllers/warehousecontroller.js

```diff
@@ -37,6 +37,32 @@ class WarehouseController extends Controller
         }
     }
 
+    removeProduct(product, customer)
+    {
+        GAME.model.warehouse.items.forEach(function (container) {
+            container.items = container.items.filter(function (item) {
+
+                if (item.name === product.name) {
```

Comment:
`==` should suffice?

Other than that, perhaps check for the negation. Nested ifs ought to be avoided :).